packages/sidecar/src/formatters/general.ts

```diff
@@ -0,0 +1,41 @@
+import type { ErrorEvent } from "@sentry/core";
+import {
+  type ParsedEnvelope,
+  type SentryEvent,
+  type SentryLogEvent,
+  type SentryTransactionEvent,
+  isErrorEvent,
+  isLogEvent,
+  isTraceEvent,
+} from "../parser/index.js";
+import human from "./human/index.js";
+import logfmt from "./logfmt/index.js";
+import type { Formatter, FormatterType } from "./types.js";
+
+const FORMATTERS: Record<FormatterType, Formatter> = {
+  human,
+  logfmt,
+  // TODO: add json formatter
+  json: logfmt, // fallback until we have a json formatter
+};
+
+export function formatEnvelope(envelope: ParsedEnvelope["envelope"], format: FormatterType): string[] {
+  const [, items] = envelope;
+  const formatter = FORMATTERS[format];
+
+  const formatted: string[] = [];
+  for (const item of items) {
+    const [{ type }, payload] = item;
+
+    if (type === "event" && isErrorEvent(payload as SentryEvent)) {
```

Comment:
This also calls for a mapping of event types to formatters as we'll have more in the future and repeated if statements are not very easy to follow or maintain.